Noisy startup when elasticsearch is unavailable

With the new HTTP client the server startup is logging very noisily when elasticsearch isn't reachable:

<details>

```

2017-05-22 15:15:14,349 ERROR: org.graylog2.periodical.BatchedElasticSearchOutputFlushThread - Caught exception while trying to flush output: {}
org.graylog2.indexer.ElasticsearchException: Couldn't check connection status of Elasticsearch
	at org.graylog2.indexer.cluster.jest.JestUtils.execute(JestUtils.java:59) ~[classes/:?]
	at org.graylog2.indexer.cluster.jest.JestUtils.execute(JestUtils.java:70) ~[classes/:?]
	at org.graylog2.indexer.cluster.Cluster.isConnected(Cluster.java:165) ~[classes/:?]
	at org.graylog2.outputs.BlockingBatchedESOutput.forceFlushIfTimedout(BlockingBatchedESOutput.java:148) ~[classes/:?]
	at org.graylog2.periodical.BatchedElasticSearchOutputFlushThread.doRun(BatchedElasticSearchOutputFlushThread.java:82) [classes/:?]
	at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:77) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_121]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_121]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: io.searchbox.client.config.exception.CouldNotConnectException: Could not connect to http://127.0.0.1:9200
	at io.searchbox.client.http.JestHttpClient.execute(JestHttpClient.java:70) ~[jest-2.4.5.jar:?]
	at org.graylog2.indexer.cluster.jest.JestUtils.execute(JestUtils.java:54) ~[classes/:?]
	... 12 more
Caused by: org.apache.http.conn.HttpHostConnectException: Connect to 127.0.0.1:9200 [/127.0.0.1] failed: Connection refused (Connection refused)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:159) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:359) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:381) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108) ~[httpclient-4.5.3.jar:4.5.3]
	at io.searchbox.client.http.JestHttpClient.executeRequest(JestHttpClient.java:132) ~[jest-2.4.5.jar:?]
	at io.searchbox.client.http.JestHttpClient.execute(JestHttpClient.java:67) ~[jest-2.4.5.jar:?]
	at org.graylog2.indexer.cluster.jest.JestUtils.execute(JestUtils.java:54) ~[classes/:?]
	... 12 more
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:1.8.0_121]
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:1.8.0_121]
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[?:1.8.0_121]
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[?:1.8.0_121]
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[?:1.8.0_121]
	at java.net.Socket.connect(Socket.java:589) ~[?:1.8.0_121]
	at org.apache.http.conn.socket.PlainConnectionSocketFactory.connectSocket(PlainConnectionSocketFactory.java:75) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:359) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:381) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) ~[httpclient-4.5.3.jar:4.5.3]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108) ~[httpclient-4.5.3.jar:4.5.3]
	at io.searchbox.client.http.JestHttpClient.executeRequest(JestHttpClient.java:132) ~[jest-2.4.5.jar:?]
	at io.searchbox.client.http.JestHttpClient.execute(JestHttpClient.java:67) ~[jest-2.4.5.jar:?]
	at org.graylog2.indexer.cluster.jest.JestUtils.execute(JestUtils.java:54) ~[classes/:?]
	... 12 more
```

</details>

For this type of error it should be enough to log single lines. Previous version have also eventually failed the graylog server start, but I'm not sure if we still want that behavior.

--- a/graylog2-server/src/main/java/org/graylog2/alerts/AlertScanner.java
+++ b/graylog2-server/src/main/java/org/graylog2/alerts/AlertScanner.java
@@ -90,7 +90,12 @@ public class AlertScanner {
                 }
             }
         } catch (Exception e) {
-            LOG.error("Skipping alert check that threw an exception.", e);
+            if (LOG.isDebugEnabled()) {
+                LOG.error("Skipping alert check <{}/{}>", alertCondition.getTitle(), alertCondition.getId(), e);
+            } else {
+                LOG.error("Skipping alert check <{}/{}>: {} ({})", alertCondition.getTitle(),
+                        alertCondition.getId(), e.getMessage(), e.getClass().getSimpleName());
+            }
         }
         return false;
     }

--- a/graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java
+++ b/graylog2-server/src/main/java/org/graylog2/indexer/cluster/Cluster.java
@@ -27,6 +27,7 @@ import io.searchbox.cluster.Health;
 import io.searchbox.cluster.NodesInfo;
 import io.searchbox.core.Cat;
 import io.searchbox.core.CatResult;
+import org.graylog2.indexer.ElasticsearchException;
 import org.graylog2.indexer.IndexSetRegistry;
 import org.graylog2.indexer.cluster.jest.JestUtils;
 import org.slf4j.Logger;
@@ -65,12 +66,21 @@ public class Cluster {
         this.requestTimeout = requestTimeout;
     }
 
-    private JsonNode clusterHealth(Collection<? extends String> indices) {
+    private Optional<JsonNode> clusterHealth(Collection<? extends String> indices) {
         final Health request = new Health.Builder()
                 .addIndex(indices)
                 .build();
-        final JestResult jestResult = JestUtils.execute(jestClient, request, () -> "Couldn't read cluster health for indices " + indices);
-        return jestResult.getJsonObject();
+        try {
+            final JestResult jestResult = JestUtils.execute(jestClient, request, () -> "Couldn't read cluster health for indices " + indices);
+            return Optional.of(jestResult.getJsonObject());
+        } catch(ElasticsearchException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.error("{} ({})", e.getMessage(), Optional.ofNullable(e.getCause()).map(Throwable::getMessage).orElse("n/a"), e);
+            } else {
+                LOG.error("{} ({})", e.getMessage(), Optional.ofNullable(e.getCause()).map(Throwable::getMessage).orElse("n/a"));
+            }
+            return Optional.empty();
+        }
     }
 
     /**
@@ -79,7 +89,7 @@ public class Cluster {
      * @return the cluster health response
      */
     public Optional<JsonNode> health() {
-        return Optional.of(clusterHealth(Arrays.asList(indexSetRegistry.getIndexWildcards())));
+        return clusterHealth(Arrays.asList(indexSetRegistry.getIndexWildcards()));
     }
 
     /**
@@ -91,7 +101,7 @@ public class Cluster {
      * @return the cluster health response
      */
     public Optional<JsonNode> deflectorHealth() {
-        return Optional.of(clusterHealth(Arrays.asList(indexSetRegistry.getWriteIndexAliases())));
+        return clusterHealth(Arrays.asList(indexSetRegistry.getWriteIndexAliases()));
     }
 
     /**
@@ -155,9 +165,16 @@ public class Cluster {
                 .timeout(Ints.saturatedCast(requestTimeout.toSeconds()))
                 .build();
 
-        final JestResult result = JestUtils.execute(jestClient, request, () -> "Couldn't check connection status of Elasticsearch");
-        final int numberOfDataNodes = result.getJsonObject().path("number_of_data_nodes").asInt();
-        return numberOfDataNodes > 0;
+        try {
+            final JestResult result = JestUtils.execute(jestClient, request, () -> "Couldn't check connection status of Elasticsearch");
+            final int numberOfDataNodes = result.getJsonObject().path("number_of_data_nodes").asInt();
+            return numberOfDataNodes > 0;
+        } catch (ElasticsearchException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.error(e.getMessage(), e);
+            }
+            return false;
+        }
     }
 
     /**

